python/lsst/afw/table/schema/schema.cc

```diff
@@ -195,6 +237,30 @@ void declareKeySpecializations(PyKey<Array<U>> &cls) {
         }
         return py::tuple(result);
     });
+    cls.def(py::pickle(
+            [](Key<Array<U>> const &self) {
+                /* Return a tuple that fully encodes the state of the object */
+                return py::make_tuple(self.getOffset(), self.getElementCount());
+            },
+            [](py::tuple t) {
+                if (t.size() != 2) throw std::runtime_error("Invalid number of parameters when unpickling!");
+                return detail::Access::makeKeyArray<U>(t[0].cast<int>(), t[1].cast<int>());
+            }));
+}
+
+void declareKeySpecializations(PyKey<std::string> &cls) {
+    declareKeyAccessors(cls);
+    cls.def_property_readonly("subfields", [](py::object const &) { return py::none(); });
+    cls.def_property_readonly("subkeys", [](py::object const &) { return py::none(); });
```

Comment:
Do you actually need these properties at all?

Reply:
I copied them from the `PyKey<T>` specialization: that was being used `string` before I added this one, so I would assume I should keep it consistent with the new specialization.